--- a/drivers/leds/Kconfig	2016-09-15 01:29:29.000000000 -0500
+++ b/drivers/leds/Kconfig	2016-09-26 02:36:55.349774421 -0500
@@ -49,6 +49,15 @@
 	help
 	 This option enables support for the LEDs on the AAT1290.

+config LEDS_APU2
+	tristate "LED Support for the PCEngines APU2 board"
+	depends on LEDS_CLASS
+	depends on GPIOLIB
+	help
+	  This option enables support for the CPU GPIO pins on the PCEngines
+	  APU2 board, as well as adds system support for the reset button and
+	  front panel LEDs.
+
 config LEDS_BCM6328
 	tristate "LED Support for Broadcom BCM6328"
 	depends on LEDS_CLASS
diff -Naur ./a/Makefile ./b/Makefile
--- a/drivers/leds/Makefile	2016-09-26 02:23:39.787307000 -0500
+++ b/drivers/leds/Makefile	2016-09-26 02:39:41.669443541 -0500
@@ -8,6 +8,7 @@
 # LED Platform Drivers
 obj-$(CONFIG_LEDS_88PM860X)		+= leds-88pm860x.o
 obj-$(CONFIG_LEDS_AAT1290)		+= leds-aat1290.o
+obj-$(CONFIG_LEDS_APU2)			+= leds-apu2.o
 obj-$(CONFIG_LEDS_BCM6328)		+= leds-bcm6328.o
 obj-$(CONFIG_LEDS_BCM6358)		+= leds-bcm6358.o
 obj-$(CONFIG_LEDS_BD2802)		+= leds-bd2802.o
