--- a/drivers/gpio/Kconfig	2016-09-15 01:29:29.000000000 -0500
+++ b/drivers/gpio/Kconfig	2016-09-26 02:29:15.714674624 -0500
@@ -307,6 +307,14 @@
 	select GPIO_GENERIC
 	select GENERIC_IRQ_CHIP

+config GPIO_NCT5104D
+	tristate "NCT5104D GPIO support"
+	depends on X86 # unconditional access to IO space.
+	select GPIO_GENERIC
+   help
+     Say yes here to support GPIO functionality of NCT5104D super I/O chip on
+     PC Engine APU boards.
+
 config GPIO_OCTEON
 	tristate "Cavium OCTEON GPIO"
 	depends on GPIOLIB && CAVIUM_OCTEON_SOC
diff -Naur ./a/Makefile ./b/Makefile
--- a/drivers/gpio/Makefile	2016-09-15 01:29:29.000000000 -0500
+++ b/drivers/gpio/Makefile	2016-09-26 02:25:57.807051231 -0500
@@ -69,6 +69,7 @@
 obj-$(CONFIG_GPIO_MVEBU)        += gpio-mvebu.o
 obj-$(CONFIG_GPIO_MXC)		+= gpio-mxc.o
 obj-$(CONFIG_GPIO_MXS)		+= gpio-mxs.o
+obj-$(CONFIG_GPIO_NCT5104D)		+= gpio-nct5104d.o
 obj-$(CONFIG_GPIO_OCTEON)	+= gpio-octeon.o
 obj-$(CONFIG_GPIO_OMAP)		+= gpio-omap.o
 obj-$(CONFIG_GPIO_PCA953X)	+= gpio-pca953x.o
