--- a/drivers/leds/Kconfig	2015-09-29 18:44:17.804056800 -0400
+++ b/drivers/leds/Kconfig	2015-09-29 19:21:36.780056800 -0400
@@ -112,6 +112,15 @@
	help
	  This option enables support for the PCEngines WRAP programmable LEDs.

+config LEDS_APU2
+	tristate "LED Support for the PCEngines APU2 board"
+	depends on LEDS_CLASS
+	depends on GPIOLIB
+	help
+	  This option enables support for the CPU GPIO pins on the PCEngines APU2 board.
+	  Note this driver still needs to be modified to use the LED class, so until then
+	  manual GPIO control only.
+
 config LEDS_COBALT_QUBE
	tristate "LED Support for the Cobalt Qube series front LED"
	depends on LEDS_CLASS
--- a/drivers/leds/Makefile	2015-09-29 18:44:17.892056800 -0400
+++ b/drivers/leds/Makefile	2015-09-29 19:11:36.572056800 -0400
@@ -56,6 +56,7 @@
 obj-$(CONFIG_LEDS_SYSCON)		+= leds-syscon.o
 obj-$(CONFIG_LEDS_VERSATILE)		+= leds-versatile.o
 obj-$(CONFIG_LEDS_MENF21BMC)		+= leds-menf21bmc.o
+obj-$(CONFIG_LEDS_APU2)			+= leds-apu2.o

 # LED SPI Drivers
 obj-$(CONFIG_LEDS_DAC124S085)		+= leds-dac124s085.o
